packages/ui/cypress/tests/watched-accounts.cy.ts

```diff
@@ -58,4 +63,154 @@ describe('Watched Accounts', () => {
+        .should('have.text', watchMultisigs['multisig-with-pure'].name)
+    })
+    // navigate to the multisig page and ensure the name is included in the home page header
+    topMenuItems.homeButton().click()
+    multisigPage.accountHeader().within(() => {
+      accountDisplay.nameLabel()
+        .should('have.text', watchMultisigs['multisig-with-pure'].name)
+    })
+  })
+
+  it('can see the identicon and badge displayed for a watched multisig', () => {
+    cy.visit(settingsPageWatchAccountUrl)
+    addWatchAccount(
+      watchMultisigs['multisig-without-pure'].address,
+      watchMultisigs['multisig-without-pure'].name)
+    // ensure the multisig badge is displayed in the selectable drop-down option
+    topMenuItems.multiproxySelector()
+      .should('be.visible')
+      .first().click()
+    topMenuItems.multiproxySelectorOption().within(() => {
+      accountDisplay.identicon().should('be.visible')
+      accountDisplay.multisigBadge().should('be.visible')
+      accountDisplay.pureBadge().should('not.exist')
+    })
+    // navigate to the home page and ensure the multisig badge is included in the header
+    topMenuItems.homeButton().click()
+    multisigPage.accountHeader().within(() => {
+      accountDisplay.identicon().should('be.visible')
+      accountDisplay.multisigBadge().should('be.visible')
+      accountDisplay.pureBadge().should('not.exist')
+    })
+  })
+
+  it('can see the identicon and badge displayed for a watched pure proxy', () => {
+    cy.visit(settingsPageWatchAccountUrl)
+    addWatchAccount(
+      watchMultisigs['multisig-with-pure'].pureAddress,
+      watchMultisigs['multisig-with-pure'].name)
+    // ensure the pure badge is displayed in the selectable drop-down option
+    topMenuItems.multiproxySelector()
+      .should('be.visible')
+      .first().click()
+    topMenuItems.multiproxySelectorOption().within(() => {
+      accountDisplay.identicon().should('be.visible')
+      accountDisplay.pureBadge().should('be.visible')
+      accountDisplay.multisigBadge().should('not.exist')
+    })
+    // navigate to the home page and ensure the pure badge is included in the header
+    topMenuItems.homeButton().click()
+    multisigPage.accountHeader().within(() => {
+      accountDisplay.identicon().should('be.visible')
+      accountDisplay.pureBadge().should('be.visible')
+      accountDisplay.multisigBadge().should('not.exist')
+    })
+  })
+
+  it('can edit the name of a watched pure', () => {
+    cy.visit(settingsPageWatchAccountUrl)
+    addWatchAccount(
+      watchMultisigs['multisig-with-pure'].pureAddress,
+      watchMultisigs['multisig-with-pure'].name)
+    // navigate to the home page and edit the name
+    topMenuItems.homeButton().click()
+    multisigPage.optionsMenuButton().click()
+    multisigPage.editNamesMenuOption().click()
+    editNamesModal.body().should('be.visible')
+    editNamesModal.inputEditPureName().type(`{selectall}{del}${`Edited Name Test`}`)
+    cy.wait(500)
```

Comment:
This is another annoying situation where this 500ms wait is needed before clicking the save button, without it the test will fail because the name in the header never updates. With this slight wait, it works every time. 

I understand we never want to never use waits but this is another instance where I couldn't go further without one. Open to suggestions.

Reply:
Let's see what we can do here. It's weird that the input are so sensitive. Maybe we can wait until the text is visible in the input or something. I'll check.

Reply:
Need to spend time on this, sorry that I didn't get to it today, will check asap and report back.

Reply:
I learned something pretty :exploding_head: today with `cy.clock` and `cy.tick`. The reason you had to wait, is because of the way we handle the storage of names. We don't store names on press, we debounce it. Without de-bouncing this field, it appeared unresponsive (I remember adding this). There's a debounce of 300ms here https://github.com/ChainSafe/Multix/blob/2d4fa20d13af25b365fe327a480b8710ffcb4f94/packages/ui/src/components/AccountEditName.tsx#L30-L31

Now thanks to `cy.clock` and `tick` we can bend time..  and make things happen quicker :shrug: 
https://docs.cypress.io/api/commands/tick

Reply:
@Lykhoyda I'll try to use this in the connect function instead of `waitUntil`. Because it is actually retrying stuff every 200ms, so it's not really elegant

Reply:
> I learned something pretty 🤯 today with `cy.clock` and `cy.tick`. The reason you had to wait, is because of the way we handle the storage of names. We don't store names on press, we debounce it. Without de-bouncing this field, it appeared unresponsive (I remember adding this). 

@Tbaut This is great! The explanation totally makes sense. Thank you for working this out, this is something I would have never considered but now we can look out for it in other scenarios! 🙏

(cc @juans-chainsafe - there's some good info in the comment above, we can look out for this in other frontend projects when encountering those scenarios where we think we need a `cy.wait`).